research-software-directory/RSD-as-a-service
# 327 & 328 organisation maintainer rls

# Organisation maintainer features for software and projects

Closes #327 
Closes #328
Related to #346
Related to #350 

Changes proposed in this pull request:
* Organisation maintainer is able to see all software affiliated with the organisation incl. not yet published software items
* Organisation maintainer is able to pin/unpin, deny/accept project affiliation with organisation, and see not yet published projects linking to organisation

How to test:
* `docker-compose down --volumes`: to remove old
* `docker-compose build`: to build all services
* `docker-compose up`: to start
* perform data migration
* start app and login as test user
* make yourself (primary) maintainer of an organisation (or all organisations)
* navigate to your organisation
* pin/unpin software, reload page to see impact, you can also logout and view organisation page as "viewer"
* login as different user
* create new software, add organisation of the "previous" maintainer
* navigate to organisation, you should not be able to see new software you just created because it is not published yet
* logout and login as "previous" user
* navigate to your organisation, you should be able to see not published software from the "second" user in the list
* create new project,
* navigate to your organisation page, you should be able to see new project, even if not published yet
* deny/approve affiliation from the organisation page
* pin/unpin project from organisation page

## Example features organisation maintainer on project page
![image](https://user-images.githubusercontent.com/9204081/174828702-04e14643-8d83-4e53-9396-51dba1abfa25.png)

PR Checklist:
